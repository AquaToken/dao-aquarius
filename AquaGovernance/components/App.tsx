import * as React from 'react';
import { lazy, Suspense, useEffect } from 'react';
import { hot } from 'react-hot-loader';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';

import Provider from '../store';
import { MainRoutes } from '../routes';
import Header, { HeaderNavLink } from '../../common/components/Header/Header';
import Footer from '../../common/components/Footer/Footer';
import AppGlobalStyle from '../../common/components/AppGlobalStyles';
import ModalContainer from '../../common/modals/atoms/ModalContainer';
import useGlobalSubscriptions from '../../common/hooks/useGlobalSubscriptions';
import ToastContainer from '../../common/toasts/ToastContainer';
import useProposalsStore from '../store/proposalsStore/useProposalsStore';
import PageLoader from '../../common/basics/PageLoader';
import ProposalCreationPage from './ProposalCreationPage/ProposalCreationPage';

const MainPage = lazy(() => import('./MainPage/MainPage'));
const VoteProposalPage = lazy(() => import('./VoteProposalPage/VoteProposalPage'));

const App = () => {
    useGlobalSubscriptions();

    const { proposals, getProposals } = useProposalsStore();

    useEffect(() => {
        getProposals();
    }, []);

    if (!proposals.length) {
        return null;
    }

    return (
        <Router>
            <Header>
                <HeaderNavLink to={MainRoutes.main}>Proposals</HeaderNavLink>
            </Header>
            <Suspense fallback={<PageLoader />}>
                <Switch>
                    <Route exact path={MainRoutes.main}>
                        <MainPage />
                    </Route>
                    <Route path={`${MainRoutes.proposal}/:id`} component={VoteProposalPage}>
                        {/*<VoteProposalPage />*/}
                    </Route>
                    <Route path={MainRoutes.create}>
                        <ProposalCreationPage />
                    </Route>
                </Switch>
            </Suspense>
            <Footer />

            <ModalContainer />
            <ToastContainer />
        </Router>
    );
};

const ProvidedApp = () => {
    return (
        <Provider>
            <AppGlobalStyle />
            <App />
        </Provider>
    );
};

declare let module: Record<string, unknown>;

export default hot(module)(ProvidedApp);
