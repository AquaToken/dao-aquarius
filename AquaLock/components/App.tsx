import * as React from 'react';
import { Suspense, useEffect } from 'react';
import { hot } from 'react-hot-loader';
import { BrowserRouter as Router, Redirect, Route, Switch } from 'react-router-dom';

import Provider from '../store';
import { MainRoutes } from '../routes';
import Header from '../../common/components/Header/Header';
import Footer from '../../common/components/Footer/Footer';
import NotFoundPage from '../../common/components/NotFoundPage/NotFoundPage';
import AppGlobalStyle from '../../common/components/AppGlobalStyles';
import ModalContainer from '../../common/modals/atoms/ModalContainer';
import useGlobalSubscriptions from '../../common/hooks/useGlobalSubscriptions';
import ToastContainer from '../../common/toasts/ToastContainer';
import PageLoader from '../../common/basics/PageLoader';
import useAuthStore from '../../common/store/authStore/useAuthStore';
import MainPage from './MainPage/MainPage';
import FAQ from './FAQ/FAQ';
import AccountPage from './AccountPage/AccountPage';
import { ModalService } from '../../common/services/globalServices';
import SnapshotPassedModal from './common/SnapshotPassedModal/SnapshotPassedModal';
import Background from '../../common/assets/img/snapshot-passed-background.svg';
import styled from 'styled-components';

const ModalBG = styled(Background)`
    object-position: center center;
`;

const App = () => {
    useGlobalSubscriptions();

    const { account } = useAuthStore();

    useEffect(() => {
        ModalService.openModal(SnapshotPassedModal, {}, true, <ModalBG />);
    }, []);

    return (
        <Router>
            <Header>
                <>
                    <a
                        href="https://aqua.network/airdrop2"
                        target="_blank"
                        rel="noreferrer noopener"
                    >
                        Airdrop
                    </a>
                </>
            </Header>
            <Suspense fallback={<PageLoader />}>
                <Switch>
                    <Route
                        exact
                        path={MainRoutes.main}
                        render={({ location }) =>
                            account ? (
                                <Redirect
                                    to={{
                                        pathname: `${MainRoutes.main}${account.accountId()}`,
                                        state: { from: location },
                                    }}
                                />
                            ) : (
                                <MainPage />
                            )
                        }
                    />

                    <Route path={`${MainRoutes.main}:accountId`}>
                        <AccountPage />
                    </Route>
                    <Route component={NotFoundPage} />
                </Switch>
            </Suspense>
            <FAQ />
            <Footer />

            <ModalContainer />
            <ToastContainer />
        </Router>
    );
};

const ProvidedApp = () => {
    return (
        <Provider>
            <AppGlobalStyle />
            <App />
        </Provider>
    );
};

declare let module: Record<string, unknown>;

export default hot(module)(ProvidedApp);
