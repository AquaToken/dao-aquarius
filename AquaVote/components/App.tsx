import * as React from 'react';
import { lazy, Suspense } from 'react';
import { hot } from 'react-hot-loader';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';

import Provider from '../store';
import { MainRoutes } from '../routes';
import Header, { HeaderNavLink } from '../../common/components/Header/Header';
import Footer from '../../common/components/Footer/Footer';
import AppGlobalStyle from '../../common/components/AppGlobalStyles';
import ModalContainer from '../../common/modals/atoms/ModalContainer';
import useGlobalSubscriptions from '../../common/hooks/useGlobalSubscriptions';
import ToastContainer from '../../common/toasts/ToastContainer';
import PageLoader from '../../common/basics/PageLoader';

const MainPage = lazy(() => import('./MainPage/MainPage'));
const AboutPage = lazy(() => import('./AboutPage/AboutPage'));

const App = () => {
    useGlobalSubscriptions();

    return (
        <Router>
            <Header>
                <>
                    <HeaderNavLink to={MainRoutes.main}>Explore</HeaderNavLink>
                    <HeaderNavLink to={MainRoutes.about}>Top 100</HeaderNavLink>
                    <HeaderNavLink to={MainRoutes.about}>About</HeaderNavLink>
                    <HeaderNavLink to={MainRoutes.about}>Airdrop</HeaderNavLink>
                </>
            </Header>
            <Suspense fallback={<PageLoader />}>
                <Switch>
                    <Route exact path={MainRoutes.main}>
                        <MainPage />
                    </Route>
                    <Route path={MainRoutes.about}>
                        <AboutPage />
                    </Route>
                </Switch>
            </Suspense>
            <Footer />

            <ModalContainer />
            <ToastContainer />
        </Router>
    );
};

const ProvidedApp = () => {
    return (
        <Provider>
            <AppGlobalStyle />
            <App />
        </Provider>
    );
};

declare let module: Record<string, unknown>;

export default hot(module)(ProvidedApp);
