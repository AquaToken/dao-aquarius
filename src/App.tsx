import * as React from 'react';
import { lazy, Suspense, useEffect, useState } from 'react';
import { hot } from 'react-hot-loader';
import { BrowserRouter as Router, Redirect, Route, Switch } from 'react-router-dom';
import useGlobalSubscriptions from './common/hooks/useGlobalSubscriptions';
import useAssetsStore from './store/assetsStore/useAssetsStore';
import useAuthStore from './store/authStore/useAuthStore';
import { StellarService, WalletConnectService } from './common/services/globalServices';
import Header, { HeaderNavLink } from './common/components/Header/Header';
import { MainRoutes } from './routes';
import PageLoader from './common/basics/PageLoader';
import NotFoundPage from './common/components/NotFoundPage/NotFoundPage';
import { Breakpoints, COLORS } from './common/styles';
import ToastContainer from './common/toasts/ToastContainer';
import { respondDown } from './common/mixins';
import Provider from './store';
import ModalContainer from './common/modals/atoms/ModalContainer';
import Footer from './common/components/Footer/Footer';
import { createGlobalStyle } from 'styled-components';
import AppGlobalStyle from './common/components/AppGlobalStyles';
import Governance from './pages/governance/Governance';
import Title from 'react-document-title';

const MainPage = lazy(() => import('./pages/main/MainPage'));
const LockerPage = lazy(() => import('./pages/locker/Locker'));
const VotePage = lazy(() => import('./pages/vote/Vote'));
const BribesPage = lazy(() => import('./pages/bribes/Bribes'));
const MarketPage = lazy(() => import('./pages/market/Market'));
const RewardsPage = lazy(() => import('./pages/rewards/Rewards'));
const AirdropPage = lazy(() => import('./pages/airdrop/Airdrop'));
const Airdrop2Page = lazy(() => import('./pages/airdrop2/Airdrop2'));
const ProfilePage = lazy(() => import('./pages/profile/Profile'));
const WalletConnectPage = lazy(() => import('./pages/wallet-connect/WalletConnect'));
const AmmPage = lazy(() => import('./pages/amm/Amm'));
const SwapPage = lazy(() => import('./pages/swap/Swap'));

const UPDATE_ASSETS_DATE = 'update assets timestamp';
const UPDATE_PERIOD = 24 * 60 * 60 * 1000;

const App = () => {
    const [wcLoginChecked, setWcLoginChecked] = useState(false);
    useGlobalSubscriptions();

    const { getAssets, assets, processNewAssets, assetsInfo, clearAssets } = useAssetsStore();
    const [isAssetsUpdated, setIsAssetsUpdated] = useState(false);
    const [balances, setBalances] = useState(null);

    const { isLogged, account, isRedirectEnabled, disableRedirect } = useAuthStore();

    useEffect(() => {
        const assetUpdateTimestamp = localStorage.getItem(UPDATE_ASSETS_DATE);

        if (!assetUpdateTimestamp || Date.now() - Number(assetUpdateTimestamp) > UPDATE_PERIOD) {
            clearAssets();
            localStorage.setItem(UPDATE_ASSETS_DATE, Date.now().toString());
            setIsAssetsUpdated(true);
        } else {
            setIsAssetsUpdated(true);
        }

        getAssets();
    }, []);

    useEffect(() => {
        WalletConnectService.loginIfSessionExist().then(() => {
            setWcLoginChecked(true);
        });
    }, []);

    useEffect(() => {
        window.addEventListener('online', reloadIfNotLoaded);

        return () => window.removeEventListener('online', reloadIfNotLoaded);
    }, [wcLoginChecked, isAssetsUpdated]);

    const reloadIfNotLoaded = () => {
        if (!wcLoginChecked || !isAssetsUpdated) {
            window.location.reload();
        }
    };

    useEffect(() => {
        if (assets.length) {
            processNewAssets(assets);
        }
    }, [assets]);

    useEffect(() => {
        if (account) {
            account.getBalances().then((res) => {
                setBalances(res);
                processNewAssets(res.map(({ asset }) => asset));
            });
            return;
        }
        setBalances(null);
    }, [account]);

    useEffect(() => {
        if (isLogged) {
            StellarService.startClaimableBalancesStream(account.accountId());
        } else {
            StellarService.closeClaimableBalancesStream();
        }
    }, [isLogged]);

    useEffect(() => {
        if (isLogged && isRedirectEnabled) {
            disableRedirect();
        }
    }, [isLogged, isRedirectEnabled]);

    useEffect(() => {
        const userAgent = window.navigator.userAgent;

        if (userAgent.match(/iPad/i) || userAgent.match(/iPhone/i)) {
            document.documentElement.style.overflowX = 'unset';
            document.body.style.overflowX = 'unset';
        }
    }, []);

    if (!isAssetsUpdated || !assetsInfo.size || !wcLoginChecked) {
        return <PageLoader />;
    }

    return (
        <Router>
            {isLogged && isRedirectEnabled && <Redirect to={MainRoutes.account} />}
            <Header>
                <>
                    {/*<HeaderNavLink*/}
                    {/*    to={MainRoutes.vote}*/}
                    {/*    exact*/}
                    {/*    activeStyle={{*/}
                    {/*        fontWeight: 700,*/}
                    {/*    }}*/}
                    {/*    title="Voting"*/}
                    {/*>*/}
                    {/*    Voting*/}
                    {/*</HeaderNavLink>*/}
                    {/*<HeaderNavLink*/}
                    {/*    to={MainRoutes.rewards}*/}
                    {/*    activeStyle={{*/}
                    {/*        fontWeight: 700,*/}
                    {/*    }}*/}
                    {/*    title="Rewards"*/}
                    {/*>*/}
                    {/*    Rewards*/}
                    {/*</HeaderNavLink>*/}
                    {/*<HeaderNavLink*/}
                    {/*    to={MainRoutes.bribes}*/}
                    {/*    activeStyle={{*/}
                    {/*        fontWeight: 700,*/}
                    {/*    }}*/}
                    {/*    title="Bribes"*/}
                    {/*>*/}
                    {/*    Bribes*/}
                    {/*</HeaderNavLink>*/}
                    {/*<HeaderNavLink*/}
                    {/*    to={MainRoutes.locker}*/}
                    {/*    activeStyle={{*/}
                    {/*        fontWeight: 700,*/}
                    {/*    }}*/}
                    {/*    title="Locker"*/}
                    {/*>*/}
                    {/*    Locker*/}
                    {/*</HeaderNavLink>*/}
                    {/*<HeaderNavLink*/}
                    {/*    to={MainRoutes.governance}*/}
                    {/*    activeStyle={{*/}
                    {/*        fontWeight: 700,*/}
                    {/*    }}*/}
                    {/*    title="Governance"*/}
                    {/*>*/}
                    {/*    Governance*/}
                    {/*</HeaderNavLink>*/}
                    {/*<HeaderNavLink*/}
                    {/*    to={MainRoutes.airdrop2}*/}
                    {/*    activeStyle={{*/}
                    {/*        fontWeight: 700,*/}
                    {/*    }}*/}
                    {/*    title="Airdrop"*/}
                    {/*>*/}
                    {/*    Airdrop*/}
                    {/*</HeaderNavLink>*/}
                    <HeaderNavLink
                        to={MainRoutes.amm}
                        activeStyle={{
                            fontWeight: 700,
                        }}
                        title="AMM"
                    >
                        AMM
                    </HeaderNavLink>
                    <HeaderNavLink
                        to={MainRoutes.swap}
                        activeStyle={{
                            fontWeight: 700,
                        }}
                        title="Swap"
                    >
                        Swap
                    </HeaderNavLink>
                </>
            </Header>
            <Suspense fallback={<PageLoader />}>
                <Switch>
                    <Route exact path={MainRoutes.main}>
                        <Title title="Liquidity management layer for Stellar | Aquarius">
                            <MainPage />
                        </Title>
                    </Route>
                    <Route path={MainRoutes.locker}>
                        <Title title="Aquarius Locker">
                            <LockerPage />
                        </Title>
                    </Route>
                    <Route path={MainRoutes.governance}>
                        <Title title="Aquarius Governance">
                            <Governance />
                        </Title>
                    </Route>
                    <Route path={MainRoutes.vote}>
                        <Title title="Aquarius Voting Tool">
                            <VotePage />
                        </Title>
                    </Route>
                    <Route path={MainRoutes.bribes}>
                        <Title title="Aquarius Bribes">
                            <BribesPage />
                        </Title>
                    </Route>
                    <Route path={MainRoutes.market}>
                        <MarketPage />
                    </Route>
                    <Route path={MainRoutes.rewards}>
                        <Title title="Aquarius Rewards">
                            <RewardsPage />
                        </Title>
                    </Route>
                    <Route path={MainRoutes.airdrop}>
                        <Title title="Aquarius Airdrop">
                            <AirdropPage />
                        </Title>
                    </Route>
                    <Route path={MainRoutes.airdrop2}>
                        <Title title="Aquarius Airdrop #2">
                            <Airdrop2Page />
                        </Title>
                    </Route>

                    <Route path={MainRoutes.account}>
                        <Title title="My Aquarius">
                            {isLogged ? <ProfilePage /> : <Redirect to={MainRoutes.main} />}
                        </Title>
                    </Route>

                    <Route path={MainRoutes.walletConnect}>
                        <Title title="WalletConnect">
                            <WalletConnectPage />
                        </Title>
                    </Route>

                    <Route path={MainRoutes.amm}>
                        <Title title="AMM Example">
                            <AmmPage balances={balances} />
                        </Title>
                    </Route>

                    <Route path={MainRoutes.swap}>
                        <Title title="Swap">
                            <SwapPage balances={balances} />
                        </Title>
                    </Route>

                    <Route component={NotFoundPage} />
                </Switch>
            </Suspense>
            <Footer />

            <ModalContainer />
            <ToastContainer />
        </Router>
    );
};

const BodyStyle = createGlobalStyle`
    ${respondDown(Breakpoints.md)`
        body {
            background-color: ${COLORS.lightGray};
        } 
    `}
`;

const ProvidedApp = () => {
    return (
        <Provider>
            <AppGlobalStyle />
            <BodyStyle />
            <App />
        </Provider>
    );
};

declare let module: Record<string, unknown>;

export default hot(module)(ProvidedApp);
