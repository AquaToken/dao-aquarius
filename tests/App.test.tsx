import * as React from 'react';
import * as renderer from 'react-test-renderer';
import App from '../src/pages/vote/Vote';
import { ReactTestRenderer, ReactTestRendererJSON } from 'react-test-renderer';

it('App is rendered', () => {
    // Render App in the document
    const component: ReactTestRenderer = renderer.create(<App />);
    const tree: ReactTestRendererJSON | ReactTestRendererJSON[] = component.toJSON();
    expect(tree).toMatchSnapshot();
});
