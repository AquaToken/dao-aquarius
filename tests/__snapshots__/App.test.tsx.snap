// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Governance is rendered 1`] = `
<div
  className="app"
>
  <h1>
    Hello World!
  </h1>
  <p>
    Foo to the barz
  </p>
  <img
    height="480"
  />
</div>
`;
